go/mysql/connection.go

```diff
@@ -47,5 +48,11 @@ func (this *ConnectionConfig) Equals(other *ConnectionConfig) bool {
 }
 
 func (this *ConnectionConfig) GetDBUri(databaseName string) string {
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", this.User, this.Password, this.Key.Hostname, this.Key.Port, databaseName)
+	var ip = net.ParseIP(this.Key.Hostname)
+	if (ip != nil) && (ip.To4() == nil) {
```

Comment:
You don't need to check `ip != nil` before calling `ip.To4()`; `To4` will return `nil` as expected.